### PR TITLE
Adding unit tests for sha256

### PR DESCRIPTION
**Pull request by korjeek**
Adding comprehensive unit tests for SHA-256 implementation.
 
 These tests verify:
  1. Basic functionality with known test vectors (e.g., "abc")
  2. Handling of large input data (4KB repeated 1000 times)
  3. Edge case with repeated single-byte input (1 million 'a' characters)
 
  The tests ensure compatibility with standard SHA-256 implementations and will help detect regressions during future code changes.

### CODE REVIEW

File: src/unit/test_sha256.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#include "../sha256.c"
```
Comment (ranshid) [inline src/unit/test_sha256.c]: why do we need to include the 'c' file?

Comment (ranshid) [inline src/unit/test_sha256.c]:
```suggestion
#include "../sha256.h"

```

**Review by ranshid — COMMENTED** [review]
Looks fine. I am not against adding these tests (maybe they can help in a future need to change the sha256 implementation...) biut can you please through some words on the top comment to explain what you are adding and why?

File: src/unit/test_sha256.c (hunk 1 above)
Comment (korjeek) [reply src/unit/test_sha256.c]: my mistake, I fixed it

**Comment by ranshid** [comment]
Thank you @korjeek I meant the top comment :) , but adding these comments in the test file also works.
I edited the top comment as well. the reason is that the top comment is used as the final commit message. it is better to have some content attached to each commit.

**Review by ranshid — APPROVED** [review]

**Comment by ranshid** [comment]
@korjeek can you PTAL at the test failures and fix?

**Comment by korjeek** [comment]
> @korjeek can you PTAL at the test failures and fix?

seems like I fixed it

**Review by ranshid — APPROVED** [review]
